src/Microsoft.Extensions.Http.Polly/PolicyHttpMessageHandler.cs

```diff
@@ -77,7 +92,13 @@ protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage reques
                 throw new ArgumentNullException(nameof(request));
             }
 
-            return _policy.ExecuteAsync((ct) => base.SendAsync(request, ct), cancellationToken);
+            var policy = ResolvePolicy(request);
+            return policy.ExecuteAsync((ct) => base.SendAsync(request, ct), cancellationToken);
+        }
+
+        private IAsyncPolicy<HttpResponseMessage> ResolvePolicy(HttpRequestMessage request)
+        {
+            return _policy ?? _policyFactory(request);
```

Comment:
@rynowak How do you plan to make to `PolicyRegistry` accessible from this class?

Reply:
I don't plan to do that. You provide a lambda that resolves the policy, take a look at the sample and let me know if that won't meet your needs.